client/components/newdeck.jsx

```diff
@@ -15,30 +15,49 @@ export default class NewDeck extends React.Component {
 
   handleSubmit(event) {
     event.preventDefault();
-    const req = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify(this.state)
-    };
-    fetch('/api/create-deck', req)
-      .then(res => {
-        res.json();
-        this.setState({ deckName: '' });
-        this.props.closeModal();
-      })
-      .catch(err => console.error(err));
+    const deckName = this.state.deckName;
+    if (deckName.length > 20) {
+      this.setState({
+        error: true,
+        errorMessage: 'Sorry, that deck name is too long!'
+      });
+    } else if (deckName.includes('?')) {
+      this.setState({
+        error: true,
+        errorMessage: "Sorry, deck names cannot contain ?'s"
+      });
+    } else {
```

Comment:
These rules seem arbitrary. I guess the length limit sort of make sense for avoiding screwing up the ui layout, but why the ? ban...?

Reply:
The ? ban is because of the parse route. When deleting the deck the request doesn't delete the deck but when viewing or adding it runs into a bug. I tested with ! and no problem so I just kicked out the ?

![image](https://user-images.githubusercontent.com/109567499/201222336-4ef5661d-2c94-4295-8f5a-b78874a72aab.png)

Reply:
I also figure most people won't try to use ?'s in their deck names BUT on the off chance they try to I do not want someone using the app to randomly find this issue in the future if I just pretended like it wasn't there. I may revise this in the future to include all special characters so that it doesn't look like I'm just targeting ?'s (even though I am)

Reply:
Are we certain that we even need the deck name in the hash params? Don't we just fetch the deck by its id anyway?

Reply:
The main reason I add the deck name to the hash params is because there are times where I want to reference the deck name without making talking to the server (this deck delete modal is an example of using the hash param instead of a request) I added the encodeURIComponent to my hrefs as you suggested and tested it out with this "Deck?"

Do you think it would be better for me to make changes to the backend by joining the tables together and then using another fetch request on pages that need to load the deck name? 

I think that the two (maybe going to be three pages) that I use the deck Name without using a fetch request are on the delete deck modal form, edit card page, and possibly going to update the hash params for the add card page.  

![encodeURIcomponent](https://user-images.githubusercontent.com/109567499/201228487-5b74d3cf-e62e-4a8b-80f8-340eca7595b5.gif)